gms/gossiper.cc

```diff
@@ -613,7 +616,10 @@ future<> gossiper::apply_state_locally(std::map<inet_address, endpoint_state> ma
             logger.trace("Ignoring gossip for {} because it is quarantined", ep);
             return make_ready_future<>();
         }
-        return seastar::with_semaphore(_apply_state_locally_semaphore, 1, [this, &ep, &map] () mutable {
+        if (_apply_state_locally_semaphore.waiters() > apply_max_queue_size) {
+            return make_exception_future<>(std::runtime_error("apply_state_locally(): max queue size reached"));
+        }
```

Comment:
I think application states are associative, no? So we can have a broadcast in progress and a pending update. If an update comes in, we do this:
1. merge the update into the _pending update
2. if there is no broadcast in progress, start one, and clear _pending
3. when the broadcast completes, start a new one if we have _pending

But I think it won't help the root cause. Probably some subscriber got stuck and doesn't make progress.

Reply:
I think coalescing updates locally can help. If propagating updates have a hard time making through, we can squash many updates by the time the next replication makes it through, reducing load on any stressed shard.

Note that situations where these updates are piling up should be transient. If they aren't we have a bigger problem at hand.

Reply:
Really, if we can not even replicate a few bytes of information to other shards in a reasonable time. The node is in a big trouble. 

Instead of playing coalescing or other tricks, I think it is better to just to ignore such updates as I mentioned here https://github.com/scylladb/scylla/pull/10968#issuecomment-1174711834.

Yes, we are not allowed to skip the gossip handler while apply the gossip state, because such handler could never be called again. But we can drop the messages we received and not respond. This is as if the message was lost. 

Even more importantly, we should fix the root cause.